pkg/lwip/contrib/netdev/lwip_netdev.c

```diff
@@ -255,22 +260,16 @@ static struct pbuf *_get_recv_pkt(netdev_t *dev)
 
 static void _event_cb(netdev_t *dev, netdev_event_t event)
 {
-    if (event == NETDEV_EVENT_ISR) {
-        assert(_pid != KERNEL_PID_UNDEF);
-        msg_t msg;
-
-        msg.type = LWIP_NETDEV_MSG_TYPE_EVENT;
-        msg.content.ptr = dev;
+    lwip_netif_t *compat_netif = dev->context;
+    assert(compat_netif != NULL);
+    struct netif *netif = &compat_netif->lwip_netif;
 
-        if (msg_send(&msg, _pid) <= 0) {
-            DEBUG("lwip_netdev: possibly lost interrupt.\n");
-        }
-    }
-    else {
-        lwip_netif_t *compat_netif = dev->context;
-        struct netif *netif = &compat_netif->lwip_netif;
-        switch (event) {
-        case NETDEV_EVENT_RX_COMPLETE: {
+    switch (event) {
+    case NETDEV_EVENT_ISR:
+        event_post(&evq, &compat_netif->ev_isr);
```

Comment:
If ev_isr happens to be queued twice, it seems that could break the list. I don't know if the drivers guarantee that doesn't happen

Reply:
The semantics of the event is that it can only be queued once. So if the event callback is executed, it means the event has been posted *at least* once since the last callback execution. So multiple fast events may trigger only a single callback execution.